helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt

```diff
@@ -199,9 +201,82 @@ private val PATH_EXCLUDE_REASON_FOR_FILENAME = listOf(
     "*.cmakein" to BUILD_TOOL_OF,
     "*.csproj" to BUILD_TOOL_OF,
     "*.gemspec" to BUILD_TOOL_OF,
+    "*.gradle" to BUILD_TOOL_OF,
+    "*.m4" to BUILD_TOOL_OF,
     "*.mk" to BUILD_TOOL_OF,
+    "*.nuspec" to BUILD_TOOL_OF,
+    "*.pdf" to DOCUMENTATION_OF,
     "*.podspec" to BUILD_TOOL_OF,
     "*.rake" to BUILD_TOOL_OF,
     "*_test.go" to TEST_OF,
-    "CMakeLists.txt" to BUILD_TOOL_OF
+    "*coverage*.sh" to BUILD_TOOL_OF,
+    ".editorconfig" to PathExcludeReason.OTHER,
+    ".ort.yml" to BUILD_TOOL_OF,
+    ".travis.yml" to BUILD_TOOL_OF,
+    "BUILD" to BUILD_TOOL_OF, // Bazel
+    "Build.PL" to BUILD_TOOL_OF,
+    "CHANGELOG*" to BUILD_TOOL_OF,
+    "CHANGES" to DOCUMENTATION_OF,
+    "CHANGES.md" to DOCUMENTATION_OF,
+    "CHANGES.txt" to DOCUMENTATION_OF,
+    "CMakeLists.txt" to BUILD_TOOL_OF,
+    "CODE_OF_CONDUCT" to DOCUMENTATION_OF,
+    "CODE_OF_CONDUCT.md" to DOCUMENTATION_OF,
+    "CONTRIBUTING" to DOCUMENTATION_OF,
+    "CONTRIBUTING.md" to DOCUMENTATION_OF,
+    "CONTRIBUTING.rst" to DOCUMENTATION_OF,
+    "CONTRIBUTING.txt" to DOCUMENTATION_OF,
+    "Cakefile" to BUILD_TOOL_OF,
+    "Cargo.toml" to BUILD_TOOL_OF,
+    "ChangeLog*" to DOCUMENTATION_OF,
+    "Configure" to BUILD_TOOL_OF,
+    "DOCS.md" to DOCUMENTATION_OF,
+    "Dockerfile" to DOCUMENTATION_OF,
+    "HISTORY.md" to DOCUMENTATION_OF,
+    "History.md" to DOCUMENTATION_OF,
+    "INSTALL" to DOCUMENTATION_OF,
+    "Makefile*" to BUILD_TOOL_OF,
+    "Makefile.*" to BUILD_TOOL_OF,
+    "NEWS.md" to DOCUMENTATION_OF,
+    "Package.swift" to BUILD_TOOL_OF,
+    "RELEASE-NOTES*" to DOCUMENTATION_OF,
+    "Rakefile*" to BUILD_TOOL_OF,
+    "SECURITY.md" to DOCUMENTATION_OF,
+    "build*.sh" to BUILD_TOOL_OF,
+    "build.gradle" to BUILD_TOOL_OF,
+    "build.rs" to BUILD_TOOL_OF, // Rust build script.
+    "build.sbt" to BUILD_TOOL_OF,
+    "changelog*" to BUILD_TOOL_OF,
+    "checksrc.bat" to BUILD_TOOL_OF,
+    "codenarc.groovy" to BUILD_TOOL_OF,
+    "conanfile.py" to BUILD_TOOL_OF,
+    "config.guess" to BUILD_TOOL_OF,
+    "config.sub" to BUILD_TOOL_OF,
+    "configure" to BUILD_TOOL_OF,
+    "configure.ac" to BUILD_TOOL_OF,
+    "depcomp" to BUILD_TOOL_OF,
+    "generate*.sh" to BUILD_TOOL_OF,
+    "gradlew.bat" to BUILD_TOOL_OF,
+    "jitpack.yml" to BUILD_TOOL_OF,
+    "make-tests.sh" to BUILD_TOOL_OF,
+    "makefile.*" to BUILD_TOOL_OF,
+    "mkdocs.yml" to BUILD_TOOL_OF,
+    "package.json" to BUILD_TOOL_OF,
+    "proguard-rules.pro" to BUILD_TOOL_OF,
+    "runsuite.c" to TEST_OF,
+    "runtest.c" to TEST_OF,
+    "settings.gradle" to BUILD_TOOL_OF,
+    "setup.cfg" to BUILD_TOOL_OF,
+    "setup.py" to BUILD_TOOL_OF,
+    "test_*.c" to BUILD_TOOL_OF
 ).checkNoDuplicatePatterns { it.first }
+
+fun main() {
+    PathExcludeGenerator.generatePathExcludes(listOf("a/b/c"))
```

Comment:
@fviernau  You adding a function here, not a pattern recommend to put this in a separate commit

Reply:
This was left over from debugging.